### ISSUE TITLE
How to initialize the keras embedding layer with pre-trained word embeddings?

### ISSUE THREAD

**Issue by zhaopku** [fchollet/keras#3306]
I'm trying to use keras to do some NLP tasks. I want to initialize the word embedding layer with pre-trained word embeddings. But how to do that?

BTW, is it mandatory that `input_dim` must be specified in `keras.layers.embeddings.Embedding(input_dim, output_dim, init='uniform', input_length=None, W_regularizer=None, activity_regularizer=None, W_constraint=None, mask_zero=False, weights=None, dropout=0.0)`

because generally you don't know how many words occur in the input data.

**Comment by mklawonn** [comment]
See <a href="https://github.com/fchollet/keras/issues/853">853</a>.

And yes input_dim is mandatory. If you are unsure about how many words occur in the input data, restricting to the top n words is common practice. You could set input_dim to the size of your word embedding vocabulary if you wanted to include everything, but this would likely be computationally prohibitive.

**Comment by YFZX** [comment]
#853 i think this issue will help you
